src/Dapr.Actors/Runtime/ActorStateManager.cs

```diff
@@ -102,8 +144,8 @@ public async Task<ConditionalValue<T>> TryGetStateAsync<T>(string stateName, Can
             {
                 var stateMetadata = stateChangeTracker[stateName];
 
-                // Check if the property was marked as remove in the cache
-                if (stateMetadata.ChangeKind == StateChangeKind.Remove)
+                // Check if the property was marked as remove in the cache or is expired
+                if (stateMetadata.ChangeKind == StateChangeKind.Remove || stateMetadata.TTLExpireTime <= DateTime.UtcNow)
```

Comment:
Nit: `DateTimeOffset.UtcNow`?

I'm virtually certain that the built-in coersion between `DateTime` and `DateTimeOffset` make it equivalent, but perhaps best to use `DateTimeOffset` as consistently as possible.

Reply:
Also might be useful to keep a watch on .NET 8 which (finally) adds a [means of abstracting time](https://learn.microsoft.com/en-us/dotnet/api/system.timeprovider?view=net-8.0) to simplify testing.

Reply:
Thanks! Done